dataPipelines/gc_thumbnails/utils.py

```diff
@@ -0,0 +1,31 @@
+from pathlib import Path
+import fitz
+
+class ThumbnailsCreator:
+    def __init__(
+        self,
+        file_name,
+        output_directory
+    ):
+        self.file_name = file_name
+        self.output_directory = output_directory
+
+    def generate_png(self):
+        p = Path(self.output_directory)
+        if not p.exists():
+            p.mkdir()
+
+        filename = str(self.file_name).split("/")[-1]
```

Comment:
if operating on string paths, use `os.path.basename(self.file_name)` to get filename and if using `pathlib.Path` you can use `Path.name`.

Reply:
Also, echoing earlier suggestion to rename `file_name` to `file_path` or `file` - here we see that confusion between whether `file_name` is just a name or path